docs/directives/needbar.rst

```diff
@@ -0,0 +1,599 @@
+.. _needbar:
+
+needbar
+========
+
+.. versionadded:: 0.7.5
+
+``needbar`` adds a bar-chart to your documentation:
+
+.. code-block:: rst
+
+   .. needbar:: My bar chart
+
+       , a, b, c
+      Z, 5,20,15
+      Y,10,15,10
+      X,15,10,20
+      W,20,15,10
+
+.. needbar:: My bar chart
+
+    , a, b, c
+   Z, 5,20,15
+   Y,10,15,10
+   X,15,10,20
+   W,20,15,10
+
+Most of the examples are using the same static data, to show the functionality of the options.
+So here is an more realistic example with data fetched from filters, together with hard coded data:
+
+.. code-block:: rst
+
+   .. needbar:: A more real bar chart
+      :legend:
+
+                   ,                           open ,                          in progress ,                          closed ,                          done ,                          implemented , number
+        Requirement,  type=='req' and status=='open', type=='req' and status=='in progress', type=='req' and status=='closed', type=='req' and status=='done', type=='req' and status=='implemented', 5
+               Test, type=='test' and status=='open',type=='test' and status=='in progress',type=='test' and status=='closed',type=='test' and status=='done',type=='test' and status=='implemented', 7
+      Specification, type=='spec' and status=='open',type=='spec' and status=='in progress',type=='spec' and status=='closed',type=='spec' and status=='done',type=='spec' and status=='implemented', 9
+
+.. needbar:: A more real bar chart
+   :legend:
+
+                ,                           open ,                          in progress ,                          closed ,                          done ,                          implemented , number
```

Comment:
Must the first line always contain the x-axis labels.

There is also ``:xlabels:``, can both be used?

I think a small paragraph about how the content must be designed is missing.

Reply:
I'm also have the feeling that we now have two ways of defining the labels.
In content or via option.

I would like to have only one, to make it a little bit easier or more clear for the user.
And then I prefer to have ``xlabels`` only, as this is already the way how other directives are asking for "column" names.
But that is just my thought, any opinions are welcome.

Reply:
Yes, you are right, both options are here. 
Reason: In the future I want to generate the complete content table during runtime with python scripting or reuse of needlist (or other new need option).
To do so, it is possible to define the labels in content or with an extra option like in the other needs. 
Maybe it is not clearly described in the description, but we need to have xlabel / ylabel. ylabel is been used as values for the legend. Even more if you are using transpose, they change the meaning. So please read x & y as columns and rows labels in the defined content data table. Maybe I have to rename them accordingly? Any opinions on?

Reply:
Your are right, I shall provide a chapter on the content itself and how it is been interpreted.

Reply:
Ok, I see here 3 scenarios, which we need to support:

1) Data with no labels for x/y and no option
2) Data with labels  in 1. row/column
3) Data with no labels, but ``x/ylabels`` option

We should support case 1), as this is the easiest way for the user to get something done with needbar.
Adding labels is for me the next step.
So we need a flag or another way to indicate, that needbar shall extract the labels from the data.

As we already have ``xlabels``, we can maybe use a magic word for this, e.g. ``FROM_DATA``.
So if ``xlabels`` is not set, not labels will be set.
``xlabels == FROM_DATA`` takes first row for labels.
``xlabels set, but != FROM_DATA``, use the value as comma separated list for the labels.

Does this makes sense for you?

Reply:
I'm currently thinking about _1. Data with no labels for x/y and no option_.
Do you like this rendering?
![need_bar_5e2f9](https://user-images.githubusercontent.com/95444300/146283324-ba3b03f5-955b-4b84-a864-1d8bf8f481d1.png)

Reply:
Yes, it's fine.
But I think the argument of ``.. needbar::`` should be also allowed to be empty, so that "Test empty label" is not shown.

I know that these (title, labels) are missing information in a good documentation, but we never know how a user wants to use it and what the first steps are to get familiar with this function. So asking for as less information as possible is helpful.

Reply:
If you give no title to the current implementation, you do not get any in the picture, so it works already fine. I will add a chapter title to the documentation, too.